universal-application-tool-0.0.1/app/services/cloud/azure/BlobStorage.java

```diff
@@ -69,9 +71,12 @@ public BlobStorage(Credentials credentials, Config config, Environment environme
   }
 
   @Override
-  public URL getPresignedUrl(String fileName) {
+  public URL getPresignedUrl(String fileName, Optional<String> prefixedOriginalFileName) {
     String blobUrl = client.getBlobUrl(fileName);
-    String sasToken = client.getSasToken(fileName);
+    // The prefixedOriginalFileName will either be in the format "dev/${filename}" or
+    // "applicant-%d/program-%d/block-%s/${filename}" where "${filename}" is the original name of
+    // the uploaded file set by a user
```

Comment:
suggestion (non-blocking): this would probably be good to move or copy to the `StorageClient` interface and add a javadoc link to the more comprehensive discussion of filenames and prefixing in the new class